### PR TITLE
UnixPB: Fix insecure downloads discovered from TrailOfBits Audit

### LINKED ISSUE #3324: Insecure software download in Ansible playbooks
Found as part of TrailOfBits Audit:

Ref: TOB-TEMURIN-4 & Appendix C

Fix the following Identified Issues:

1) infrastructure/ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/yum/tasks/main.yml
  get-url-validate-certs-disabled
  Found file download with SSL verification disabled
  
2) infrastructure/ansible/playbooks/AdoptOpenJDK_ITW_Playbook/roles/Common/tasks/CentOS.yml
yum-validate-certs-disabled
Found yum with SSL verification disabled

3) infrastructure/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/CentOS.yml
yum-validate-certs-disabled
Found yum with SSL verification disabled

4) infrastructure/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Debian.yml
apt-key-unencrypted-url
Found apt key download with unencrypted URL (e.g. HTTP, FTP, etc.)

5) infrastructure/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/SLES.yml
get-url-validate-certs-disabled
Found file download with SSL verification disabled

6) infrastructure/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Ubuntu.yml
apt-key-unencrypted-url
Found apt key download with unencrypted URL (e.g. HTTP, FTP, etc.)

7) infrastructure/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml
get-url-validate-certs-disabled
Found file download with SSL verification disabled

8) infrastructure/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml
unarchive-validate-certs-disabled
Found unarchive download with SSL verification disabled

### PR DESCRIPTION
**Pull request by steelhead31**
Fix issues identified in ToB #4

Fixes #3324 

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)

### CODE REVIEW

**Comment by steelhead31** [comment]
VPC in progress: [https://ci.adoptium.net/job/VagrantPlaybookCheck/1827/](https://ci.adoptium.net/job/VagrantPlaybookCheck/1827/)

All changes have been tested locally and have worked ok.

**Review by Haroon-Khel — APPROVED** [review]

**Review by sxa — APPROVED** [review]
Looks probably ok, although the SLES ones won't have been tested and we should verify that they will work on a SLES image first since the certs on the older SLES may have problems. Hadn't realised we had a download of an IBM Java on SLES11... Presumably because that version needed an older glibc ...

**Comment by steelhead31** [comment]
I'll run some tests on a SLES docker image, I did manage a test on a SLES 12.3 x64 vagrant box, with a custom written playbook, that simply downloads the file with those options, but I'll try running the playbook on some additional docker SLES12 & SLES15 s390x containers or similar.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@
     dest: /tmp/
     remote_src: yes
     mode: 0755
-    validate_certs: False
+    validate_certs: True
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml:33]: Others have all lowercase true - is this deliberate?

**Review by karianna — APPROVED** [review]

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml (hunk 1 above)
Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml]: Not entirely, I've just replaced and matched what was there, I'll switch it to lowercase to align with elsewhere.

**Comment by steelhead31** [comment]
Tests on SLES12 & SLES15 docker containers ok for the changed element of the Java download, for SLES12, there are many other issues related to the lack of official repositories, but these are known issues, with SLES12 being end of life.

**Pull request merged**